File: plugins/cocoaui/HelpWindowController.m
//
//  HelpWindowController.m
//  DeaDBeeF
//
//  Created by Oleksiy Yakovenko on 6/12/19.
//  Copyright © 2019 Oleksiy Yakovenko. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "HelpWindowController.h"

@interface HelpWindowController ()

@property (unsafe_unretained) IBOutlet NSTextView *textView;

@end

@implementation HelpWindowController

- (void)windowDidLoad {
    [super windowDidLoad];

    [self loadContent];
}

- (void)loadContent {
    NSURL *path = self.contentURL;
    NSString *content = [NSString stringWithContentsOfFile:path.path encoding:NSUTF8StringEncoding error:nil];
    if (content) {

        NSFont *font;

        if (@available(macOS 10.15, *)) {
            font = [NSFont monospacedSystemFontOfSize:0 weight:NSFontWeightRegular];
        } else {
            font = [NSFont systemFontOfSize:0];
        }

        [self.textView.textStorage setAttributedString:[[NSAttributedString alloc] initWithString:content attributes:@{
            NSForegroundColorAttributeName: NSColor.controlTextColor,
            NSFontAttributeName: font

        }]];
        self.textView.selectedRange = NSMakeRange(0, 0);
    }
}

- (void)setContentURL:(NSURL *)contentURL {
    _contentURL = contentURL;

    if (self.textView != nil) {
        [self loadContent];
    }
}

@end

File: plugins/cocoaui/HelpWindowController.swift
// This help window controller is a proof of concept / test case for supporting Swift coding in deadbeef codebase.
// It doesn't have any dependencies, and is very simple to replace with ObjC if any problem occur.

import Cocoa

class HelpWindowController: NSWindowController {

    @IBOutlet var textView: NSTextView!
    override func windowDidLoad() {
        super.windowDidLoad()

        if let path = Bundle.main.url(forResource: "help-cocoa", withExtension: "txt") {
            do {
                let content = try String(contentsOf: path, encoding: .utf8)
                if let textStorage = textView.textStorage {
                    textStorage.setAttributedString(NSAttributedString(string: content, attributes: [NSAttributedString.Key.foregroundColor : NSColor.controlTextColor]))
                    textView.setSelectedRange(NSMakeRange(0, 0))
                }
            }
            catch {
            }
        }
    }

}
